app/src/main/java/com/github/se/project/ui/lesson/RequestedLessons.kt

```diff
@@ -134,75 +147,147 @@ fun RequestedLessonsScreen(
+                if (selectedSubject != null) {
+                    FilterChip(
+                        selected = true,
+                        onClick = { selectedSubject = null },
+                        label = { Text(selectedSubject?.name ?: "All Subjects") },
+                        leadingIcon = { Icon(Icons.Default.Clear, "Clear filter") },
+                        modifier = Modifier.padding(horizontal = 16.dp),
+                        colors = FilterChipDefaults.filterChipColors(
+                            selectedContainerColor = MaterialTheme.colorScheme.secondaryContainer,
+                            selectedLabelColor = MaterialTheme.colorScheme.onPrimary,
+                            selectedLeadingIconColor = MaterialTheme.colorScheme.onPrimary
+                        )
+                    )
+                }
+
+                if (filteredLessonsWithScores.isEmpty()) {
+                    EmptyState(showInstants)
+                } else {
+                    LazyColumn(
+                        modifier = Modifier.fillMaxSize().testTag("lessonsList"),
+                        contentPadding = PaddingValues(horizontal = 16.dp),
+                        verticalArrangement = Arrangement.spacedBy(8.dp)
+                    ) {
+                        items(filteredLessonsWithScores.size) { index ->
+                            val (lesson, score) = filteredLessonsWithScores[index]
+                            if (!showInstants.value) {
+                                DisplayLessons(
+                                    lessons = listOf(lesson),
+                                    isTutor = (currentProfile?.role == Role.TUTOR),
+                                    onCardClick = { lessonClicked ->
+                                        lessonViewModel.selectLesson(lessonClicked)
+                                        navigationActions.navigateTo(Screen.TUTOR_LESSON_RESPONSE)
+                                    },
+                                    listProfilesViewModel = listProfilesViewModel,
+                                    requestedScreen = true,
+                                    suitabilityScore = score
+                                )
+                            } else {
+                                DisplayLessons(
+                                    lessons = listOf(lesson),
+                                    isTutor = (currentProfile?.role == Role.TUTOR),
+                                    onCardClick = { lessonClicked ->
+                                        lessonViewModel.selectLesson(lessonClicked)
+                                        navigationActions.navigateTo(Screen.TUTOR_LESSON_RESPONSE)
+                                    },
+                                    listProfilesViewModel = listProfilesViewModel,
+                                    requestedScreen = true,
+                                    distance = calculateDistance(
```

Comment:
as I said before, it makes sense to filter by distance in the case of instant lesson. nice.